test/unit/org/apache/cassandra/net/SensorsCustomParamsTest.java

```diff
@@ -95,4 +118,72 @@ public void testIndexReadBytesTableParam()
         String actualParam = SensorsCustomParams.INDEX_READ_BYTES_TABLE;
         assertEquals(expectedParam, actualParam);
     }
+
+    @Test
+    public void testAddWriteSensorToResponse()
```

Comment:
Since the two tests are pretty much identical I was wondering if we can extract most of the logic from them and do something similar to the following:

```dtd
    public void testAddWriteSensorToResponse()
    {
        testAddSensorToResponse(Type.WRITE_BYTES,
               c -> SensorsCustomParams.encodeTableInWriteBytesRequestParam(c.getTable()),
               c -> SensorsCustomParams.encodeTableInWriteBytesTableParam(c.getTable()));
    }

    @Test
    public void testAddReadSensorToResponse()
    {
        testAddSensorToResponse(Type.READ_BYTES,
                ignored -> SensorsCustomParams.READ_BYTES_REQUEST,
                ignored -> SensorsCustomParams.READ_BYTES_TABLE);
    }

    private void testAddSensorToResponse(Type sensorType, Function<Context, String> requestParamFunc, Function<Context, String> tableParamContext)
    {
        RequestSensors sensors = new RequestSensors();
        UUID tableId = UUID.randomUUID();
        KeyspaceMetadata ksm = KeyspaceMetadata.create("ks1", null);
        TableMetadata tm = TableMetadata.builder("ks1", "t1", TableId.fromString(tableId.toString()))
                                        .addPartitionKeyColumn("pk", AsciiType.instance)
                                        .build();
        SensorsRegistry.instance.onCreateKeyspace(ksm);
        SensorsRegistry.instance.onCreateTable(tm);

        Context context = new Context("ks1", "t1", tableId.toString());
        sensors.registerSensor(context, sensorType);
        sensors.incrementSensor(context, sensorType, 13.0);
        sensors.syncAllSensors();

        Message.Builder<NoPayload> builder =
        Message.builder(Verb._TEST_1, noPayload)
               .withId(1);

        switch (sensorType)
        {
            case READ_BYTES:
                SensorsCustomParams.addReadSensorToResponse(builder, sensors, context);
                break;
            case WRITE_BYTES:
                SensorsCustomParams.addWriteSensorToResponse(builder, sensors, context);
                break;
            default:
                throw new IllegalArgumentException("Unexpected sensor type " + sensorType);
        }

        Message<NoPayload> msg = builder.build();
        assertNotNull(msg.header.customParams());
        assertEquals(2, msg.header.customParams().size());
        String requestParam = requestParamFunc.apply(context);
        String tableParam = tableParamContext.apply(context);
        assertTrue(msg.header.customParams().containsKey(requestParam));
        assertTrue(msg.header.customParams().containsKey(tableParam));
        double epsilon = 0.000001;
        assertEquals(13.0, SensorsCustomParams.sensorValueFromBytes(msg.header.customParams().get(requestParam)), epsilon);
        assertEquals(13.0, SensorsCustomParams.sensorValueFromBytes(msg.header.customParams().get(tableParam)), epsilon);
    }
```

Reply:
I usually the "Rule of Three" for code duplication but really no strong opinions here.